src/lcm/LifeCycleStates.cc

```diff
@@ -2496,3 +2673,500 @@ void LifeCycleManager::trigger_resize_failure(int vid)
         Quotas::quota_del(Quotas::VM, vm_uid, vm_gid, &deltas);
     });
 }
+
+/* -------------------------------------------------------------------------- */
+/* -------------------------------------------------------------------------- */
+
+void LifeCycleManager::trigger_disk_restore_success(int vid)
+{
+    trigger([this, vid] {
+        Template vm_quotas_snp;
+        vector<Template *> ds_quotas_snp;
+        int uid = -1, gid = -1;
+
+        if ( auto vm = vmpool->get(vid) )
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This snippet may contain an accidental use of incorrect operation. Make sure that the operation being performed is in fact what you intended.